lib/service.js

```diff
@@ -74,4 +75,23 @@ export function createService(resolver, token, shutdown) {
         process.stderr.write = stderr_write;
       });
   };
+
+  /**
+   * @param {Socket} con
+   * @param {string} command
+   */
+  function onCommand(con, command) {
+    switch (command) {
+      case SHUTDOWN_COMMAND:
+        shutdown();
+        break;
+      case PING_COMMAND:
+        con.write(JSON.stringify({ isAlive: true }));
+        break;
+      default:
+        console.warn(`Unknown command: ${command}`);
```

Comment:
Unfortunately, console logs don't go anywhere on the server side. I'd write the error response to the connection here. Can only happen if someone is playing around. At least they see a response then.